src/Parallel/ResourceInfo.hpp

```diff
@@ -426,6 +496,12 @@ struct ResourceInfo {
   /// `allocate_array` function of the array component
   const std::unordered_set<size_t>& procs_to_ignore() const;
 
+  /// Returns a `std::set<size_t>` that has all processors with elements on them
+  const std::set<size_t>& procs_with_elements() const;
+
+  /// Returns the first proc that has elements on it.
+  size_t first_proc_with_elements() const;
```

Comment:
not sure if this function is useful at all, and since it is trivial to get from the previous function, I'd just get rid of it